### PR TITLE
[Google for Woo] Logic to choose which webview to use based on Jetpack connection

### PR DESCRIPTION
**Pull request by hafizrahman**
part of: #11917

### Description

Depending on the connection used in the app, we want to use two different webviews to open the Google Ads:
1. Connection with WordPress.com will use `WPComWebView`, which can automatically log in the user and redirect to the Google Ads page.
2. Connection with Connection Package*/Application Password  will use `ExitAwareWebView`, which does not automatically log in and instead present a .org login form. This will only need to be done once, though, and if the merchant opens the webview again, they will be logged in already.

* While Connection Package can technically use `WPComWebView`, in our tests the redirect doesn't work correctly and always go to wordpress.com instead. I'm unsure why this is, but seems something server-side, and for now we'll opt to not do automatic login for it.

### Steps to reproduce

1. Test with a site on WordPress.com,
2. Use the Google for WooCommerce button in More Menu. The webview should automatically log you in and open the Google Ads page,
3. Switch to a site connected with application password/Jetpack CP,
4. Use the **Google for WooCommerce** button in More Menu. The webview should open a .org login form,
5. Login and ensure it opens the Google Ads page,
6. Close the webview and use the **Google for WooCommerce** button again,
7. Ensure that it opens the Google Ads page immediately, no login needed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt [hunk 1]
```diff
@@ -194,13 +180,32 @@ class MoreMenuFragment : TopLevelFragment() {
...
+    }
+
     private fun openInAuthBrowser(url: String) {
         findNavController().navigateSafely(
             NavGraphMainDirections.actionGlobalWPComWebViewFragment(
                 urlToLoad = url,
-                urlsToTriggerExit = arrayOf(),
+                urlsToTriggerExit = arrayOf(), // todo-11917: Replace with the right success URL
                 title = getString(R.string.more_menu_button_google),
-                clearCache = true
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt:203]: this was just for testing in the previous PR and we want to keep the cache in real life.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt [hunk 2]
```diff
@@ -194,13 +180,32 @@ class MoreMenuFragment : TopLevelFragment() {
...
+                urlComparisonMode = ExitAwareWebViewViewModel.UrlComparisonMode.PARTIAL
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt]: This should be `WPComWebViewViewModel.UrlComparisonMode.PARTIAL` instead? 
```suggestion
                urlComparisonMode = WPComWebViewViewModel.UrlComparisonMode.PARTIAL
```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt [hunk 3]
```diff
@@ -194,13 +180,32 @@ class MoreMenuFragment : TopLevelFragment() {
         )
     }
 
+    private fun openGoogleForWooWebview(url: String, canAutoLogin: Boolean) {
+        if (canAutoLogin) {
+            openInAuthBrowser(url)
+        } else {
+            openInExitAwareWebview(url)
+        }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt]: Np, and probably personal preference but it usually looks nicer and less lines of code to use when style. No need to change anything, just leaving the suggestion.
```suggestion
when {
    canAutoLogin -> openInAuthBrowser(url)
    else -> openInExitAwareWebview(url)
}
```

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Nice job @hafizrahman it works as expected! Just left a couple minor suggestion. One of them should be addressed for the code to compile. Other than that :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt]: Good catch, thanks!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt (hunk 3 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuFragment.kt]: Yup, looks better.

**Pull request merged**